### PR TITLE
76614 qa automation framework setup

### PR DESCRIPTION
**Pull request by Sadha1501493**
Added basic automation framework skeleton.

### CODE REVIEW

File: UKHO.ERPFacade.API.FunctionalTests/UKHO.ERPFacade.API.FunctionalTests.csproj
Comment (jrippington) [inline UKHO.ERPFacade.API.FunctionalTests/UKHO.ERPFacade.API.FunctionalTests.csproj:1]: Please move the UKHO.ERPFacade.FunctionalTests project from the project root directory to the 'tests' folder

File: UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/WebhookScenarios.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        [SetUp]
+        public void Setup()
+        {
+            webhook = new WebhookEndpoint();
+            TestConfig = new TestConfiguration();
+            DirectoryInfo dir = Directory.GetParent(Environment.CurrentDirectory).Parent.Parent;
+            filePathWebhook = Path.Combine(dir.FullName, TestConfig.PayloadFolder, TestConfig.WebhookPayloadFileName);
+        }
+        [Test]
+        public void WhenValidEventInnewenccontentpublishedeventoptions_ThenWebhookReturns200OkResponse()
```
Comment (jrippington) [inline UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/WebhookScenarios.cs:25]: Method  signature should be public async Task  WhenValidEventInnewenccontentpublishedeventoptions_ThenWebhookReturns200OkResponse()

File: UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/WebhookScenarios.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        {
+            var response = webhook.OptionWebhookResponseAsync().Result;
+            response.StatusCode.Should().Be((System.Net.HttpStatusCode)200);
+        }
+        
+        [Test]
+        public void WhenValidEventInNewEncContentPublishedEventReceived_ThenWebhookReturns200OkResponse()
+        {
+            var response = webhook.PostWebhookResponseAsync(filePathWebhook).Result;
```
Comment (jrippington) [inline UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/WebhookScenarios.cs:34]: See above, use await

File: UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        private RestClient client;
+        public WebhookEndpoint()
+        {
+            var options = new RestClientOptions(_config.erpfacadeConfig.BaseUrl);
+            client = new RestClient(options);
+        }
+
+        public async Task<RestResponse> OptionWebhookResponseAsync()
+        {
+            String requestBody = "{ }";
```
Comment (jrippington) [inline UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs:18]: Use var

**Review by jrippington — CHANGES_REQUESTED** [review]

File: UKHO.ERPFacade.API.FunctionalTests/UKHO.ERPFacade.API.FunctionalTests.csproj
Comment (Sadha1501493) [reply UKHO.ERPFacade.API.FunctionalTests/UKHO.ERPFacade.API.FunctionalTests.csproj:1]: Done

File: UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/WebhookScenarios.cs (hunk 1 above)
Comment (Sadha1501493) [reply UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/WebhookScenarios.cs:25]: Included in PR#15

File: UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/WebhookScenarios.cs (hunk 2 above)
Comment (Sadha1501493) [reply UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/WebhookScenarios.cs:34]: Included in PR#15

File: UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs (hunk 3 above)
Comment (Sadha1501493) [reply UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs:18]: Included in PR#15

**Comment by Sadha1501493** [comment]
All comments are included in PR#15

**Pull request closed without merging**